src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -256,21 +256,42 @@ public BigInteger(decimal value)
         {
         }
 
-        public BigInteger(ReadOnlySpan<byte> value)
+        public BigInteger(ReadOnlySpan<byte> value, bool isUnsigned=false, bool isBigEndian=false)
         {
             int byteCount = value.Length;
```

Comment:
This was pre-existing, but separately/subsequently it might be interesting to look to see whether this caching helps or hurts... it could be inhibiting bounds-check removal by the JIT.

Reply:
(This may even be more relevant for some of the new paths added here, as they appear to be more aligned with the kinds of patterns the JIT recognizes for bounds-check removal.)

Reply:
The only obvious place that could benefit is in figuring out how many leading 0x00 bytes are present in a big endian value.  Everywhere else the value was being mutated, or used to seed the parallel index.

Before it could be inlined the little-endian trim would need to slice instead of just reducing the byteCount value.

So I think I'll leave this one for a followup perf tuning.  Especially since my machine is acting up and not giving consistent perf data run to run.